### PR TITLE
grunt task 'kill-processes' for node-adapter

### PR DESCRIPTION
**Pull request by andreGarvin**
@rdepena @whyn07m3 @datamadic

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -80,6 +81,33 @@ module.exports = function(grunt) {
...
+                  // return the process ids of each process found
+                  processList = processList.map( i => {
+
+                        return pid: i.pid
+                  }).forEach( process => {
+
+                      // kill each process
+                      ps.kill(process.pid)
+                  })
+              }
```
Comment (whyn07m3) [inline Gruntfile.js]: Less code: `processList.forEach(process => ps.kill(process.pid));`
No need for `if (processList.length !== 0) {...}` check and `.map` in this case.

Comment (andreGarvin) [inline Gruntfile.js]: okay

**Comment by andreGarvin** [comment]
fixed the conflict @whyn07m3

File: Gruntfile.js [hunk 2]
```diff
@@ -80,6 +81,26 @@ module.exports = function(grunt) {
...
+
+                  // if error occurs then thow error
+                  throw new Error( err );
+              }
+
+              // return the process ids of each process found
+              processList = processList.map( i => {
+		 
+		 return i.pid
+	      }).forEach( pid => ps.kill(pid))
```
Comment (whyn07m3) [inline Gruntfile.js]: This still need to be fixed, per my previous comment

Comment (andreGarvin) [inline Gruntfile.js]: okay I see what you mean

File: Gruntfile.js [hunk 3]
```diff
@@ -80,6 +81,23 @@ module.exports = function(grunt) {
         finRepl.startRepl();
     });
 
+    grunt.registerTask('kill-processes', function() {
+
+          // looks up the process by the program name
+          ps.lookup({
+              command: 'openfin.exe'
+          }, function(err, processList) {
```
Comment (whyn07m3) [inline Gruntfile.js]: should just be an arrow function `}, (err, processList) => {`

File: Gruntfile.js [hunk 4]
```diff
@@ -80,6 +81,23 @@ module.exports = function(grunt) {
...
+              if (err) {
+
+                  // if error occurs then thow error
+                  throw new Error( err );
```
Comment (whyn07m3) [inline Gruntfile.js]: extra spaces `throw new Error(err);`

Comment (whyn07m3) [inline Gruntfile.js]: Unnecessary comment

File: Gruntfile.js [hunk 5]
```diff
@@ -80,6 +81,23 @@ module.exports = function(grunt) {
...
+              }
+
+              // return the process ids of each process found
+              processList.forEach( i => ps.kill(i.pid))
+          })
```
Comment (whyn07m3) [inline Gruntfile.js]: missing semicolon

File: Gruntfile.js [hunk 6]
```diff
@@ -80,6 +81,23 @@ module.exports = function(grunt) {
...
+    })
```
Comment (whyn07m3) [inline Gruntfile.js]: missing semicolon

File: Gruntfile.js [hunk 7]
```diff
@@ -80,6 +81,23 @@ module.exports = function(grunt) {
...
+              command: 'openfin.exe'
+          }, function(err, processList) {
+              if (err) {
+
+                  // if error occurs then thow error
+                  throw new Error( err );
+              }
+
+              // return the process ids of each process found
+              processList.forEach( i => ps.kill(i.pid))
```
Comment (whyn07m3) [inline Gruntfile.js]: extra space, missing semicolon and confusing argument name (`i` is usually used for indices) `processList.forEach(process => ps.kill(process.pid));`

**Comment by whyn07m3** [comment]
not all the comments were addressed

**Comment by andreGarvin** [comment]
I made the changes @whyn07m3

File: Gruntfile.js [hunk 8]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
...
+              command: 'openfin.exe'
+          }, (err, processList) => {
+              if (err) throw new Error( err );
+              processList.forEach( i => ps.kill(i.pid));
+          });
+    });
+
+    grunt.registerTask('publish-docs', function() {
+        exec('cd docs && git commit -am "committed new update for node-adapter docuemtation github pages" && git push origin master', function(err, stdout, stderr) {
+             if ( err ) return console.log( err )
```
Comment (whyn07m3) [inline Gruntfile.js]: unnecessary spaces

File: Gruntfile.js [hunk 9]
```diff
@@ -1,6 +1,8 @@
 const path = require('path');
 const testAppConfig = path.join('test','app.json');
 const liveServer = require('live-server');
+const ps = require('ps-node');
+const exec = require("child_process").exec;
```
Comment (whyn07m3) [inline Gruntfile.js]: single quotes

File: Gruntfile.js [hunk 10]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
         finRepl.startRepl();
     });
 
+    grunt.registerTask('kill-processes', function() {
+          ps.lookup({
+              command: 'openfin.exe'
+          }, (err, processList) => {
+              if (err) throw new Error( err );
```
Comment (whyn07m3) [inline Gruntfile.js]: extra spaces

File: Gruntfile.js [hunk 11]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
...
+              processList.forEach( i => ps.kill(i.pid));
```
Comment (whyn07m3) [inline Gruntfile.js]: extra space

File: Gruntfile.js [hunk 12]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
...
+          });
+    });
+
+    grunt.registerTask('publish-docs', function() {
+        exec('cd docs && git commit -am "committed new update for node-adapter docuemtation github pages" && git push origin master', function(err, stdout, stderr) {
```
Comment (whyn07m3) [inline Gruntfile.js]: arrow function

File: Gruntfile.js [hunk 13]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
...
+             if ( err ) return console.log( err )
+             console.log("published new documentationfor node-adapter.");
```
Comment (whyn07m3) [inline Gruntfile.js]: single quotes

File: Gruntfile.js [hunk 14]
```diff
@@ -60,7 +62,8 @@ module.exports = function(grunt) {
             launch: {
                 open: true
             }
-        }
+        },
+	'publish-docs': {}
```
Comment (whyn07m3) [inline Gruntfile.js]: needs indentation. btw, what is it here for?

File: Gruntfile.js [hunk 15]
```diff
@@ -80,6 +83,22 @@ module.exports = function(grunt) {
...
+              command: 'openfin.exe'
+          }, (err, processList) => {
+              if (err) throw new Error(err);
+              processList.forEach(i=> ps.kill(i.pid));
+          });
+    });
+
+    grunt.registerTask('publish-docs', () => {
+        exec('cd docs && git commit -am "committed new update for node-adapter docuemtation github pages" && git push origin master', function(err, stdout, stderr) {
+             if (err) return console.log(err)
```
Comment (whyn07m3) [inline Gruntfile.js]: semicolon

File: Gruntfile.js (hunk 14 above)
Comment (andreGarvin) [inline Gruntfile.js]: sorry forgot to remove that @whyn07m3

File: Gruntfile.js [hunk 16]
```diff
@@ -80,6 +82,22 @@ module.exports = function(grunt) {
...
+              command: 'openfin.exe'
+          }, (err, processList) => {
+              if (err) throw new Error(err);
+              processList.forEach(i=> ps.kill(i.pid));
+          });
+    });
+
+    grunt.registerTask('publish-docs', () => {
+        exec('cd docs && git commit -am "committed new update for node-adapter docuemtation github pages" && git push origin master', function(err, stdout, stderr) {
+             if (err) return console.log(err);
```
Comment (rdepena) [inline Gruntfile.js]: lets add some {} here.

File: Gruntfile.js [hunk 17]
```diff
@@ -80,6 +82,22 @@ module.exports = function(grunt) {
         finRepl.startRepl();
     });
 
+    grunt.registerTask('kill-processes', function() {
+          ps.lookup({
+              command: 'openfin.exe'
+          }, (err, processList) => {
+              if (err) throw new Error(err);
```
Comment (rdepena) [inline Gruntfile.js]: needs {};

Comment (andreGarvin) [inline Gruntfile.js]: what does '{}' mean ? Not clear on what you are asking.

**Comment by rdepena** [comment]
👍

**Pull request merged**